src/linear_algebra/affine_space/combination.lean

```diff
@@ -229,6 +229,30 @@ lemma affine_combination_vsub (w₁ w₂ : ι → k) (p : ι → P) :
   s.affine_combination p w₁ -ᵥ s.affine_combination p w₂ = s.weighted_vsub p (w₁ - w₂) :=
 by rw [←affine_map.linear_map_vsub, affine_combination_linear, vsub_eq_sub]
 
+lemma affine_combination_attach (s : finset P) (w : P → k) :
+  s.attach.affine_combination (coe : s → P) (w ∘ coe) = s.affine_combination id w :=
+begin
+  simp only [affine_combination, weighted_vsub_of_point_apply, id.def, vadd_right_cancel_iff,
+    function.comp_app, affine_map.coe_mk],
+  let f : s → V := λ i, w (i : P) • ((i : P) -ᵥ classical.choice S.nonempty),
+  let g : P → V := λ i, w i • (i -ᵥ classical.choice S.nonempty),
+  change univ.sum f = s.sum g,
+  have hfg : f = g ∘ (coe : s → P), { ext, simp, },
+  rw hfg,
+  exact sum_attach,
+end
+
+omit S
+
+/-- Viewing a module as an affine space modelled on itself, affine combinations are just linear
+combinations. -/
+@[simp] lemma affine_combination_module
```

Comment:
In my opinion, the name shouldn't mention the typeclass. What about a plain `affine_combination_eq_linear_combination`?